test/multiplex_test.rb

```diff
@@ -0,0 +1,51 @@
+require_relative 'test_helper'
+
+class GraphQL::MultiplexTest < Minitest::Test
+  attr_reader :queries
+
+  def setup
+    @queries = []
+    QueryNotifier.subscriber = ->(query) { @queries << query }
+  end
+
+  def teardown
+    QueryNotifier.subscriber = nil
+  end
+
+  def schema_multiplex(*args)
+    ::Schema.multiplex(*args)
+  end
+
+
+  def test_batched_find_by_id
+    query_string = <<-GRAPHQL
+      query FetchTwoProducts($id1: ID!, $id2: ID!){
+        first: product(id: $id1) { id, title }
+        second: product(id: $id2) { id, title }
+      }
+    GRAPHQL
+
+    results = schema_multiplex([
+      { query: query_string, variables: { "id1" => "1", "id2" => "2"} },
+      { query: query_string, variables: { "id1" => "1", "id2" => "3"}},
+    ])
+
+    expected_1 = {
+      "data" => {
+        "first" => { "id" => "1", "title" => "Shirt" },
+        "second" => { "id" => "2", "title" => "Pants" },
+      }
+    }
+
+    expected_2 = {
+      "data" => {
+        "first" => { "id" => "1", "title" => "Shirt" },
+        "second" => { "id" => "3", "title" => "Sweater" },
+      }
+    }
+
+    assert_equal expected_1, results.first
+    assert_equal expected_2, results.last
+    assert_equal ["Product/1,2,3"], queries
+  end
```

Comment:
I tried making the following changes to the test

```diff
diff --git a/test/multiplex_test.rb b/test/multiplex_test.rb
index c120614..6748aab 100644
--- a/test/multiplex_test.rb
+++ b/test/multiplex_test.rb
@@ -24,9 +24,15 @@ class GraphQL::MultiplexTest < Minitest::Test
         second: product(id: $id2) { id, title }
       }
     GRAPHQL
+    mutation_string = <<-GRAPHQL
+      mutation {
+        counter_loader
+      }
+    GRAPHQL
 
     results = schema_multiplex([
       { query: query_string, variables: { "id1" => "1", "id2" => "2"} },
+      { query: mutation_string, context: { counter: 0 } },
       { query: query_string, variables: { "id1" => "1", "id2" => "3"}},
     ])
 
@@ -39,13 +45,20 @@ class GraphQL::MultiplexTest < Minitest::Test
 
     expected_2 = {
       "data" => {
+        "counter_loader" => 0,
+      }
+    }
+
+    expected_3 = {
+      "data" => {
         "first" => { "id" => "1", "title" => "Shirt" },
         "second" => { "id" => "3", "title" => "Sweater" },
       }
     }
 
     assert_equal expected_1, results.first
-    assert_equal expected_2, results.last
+    assert_equal expected_2, results[1]
+    assert_equal expected_3, results.last
     assert_equal ["Product/1,2,3"], queries
   end
 end
```

and it still passed, even though there is a mutation in between the queries.

What should be the expectations around the ordering of the operations?  The [graphql-ruby multiplex documentation](http://graphql-ruby.org/queries/multiplex.html) says it will "run queries concurrently", but also says this feature is for [Apollo Client’s query batching](http://dev.apollodata.com/core/network.html#query-batching) which "is a transport-level mechanism" where I would expect the order of operations would be preserved.